### PR TITLE
Add unit test for surface.get_losses

### PR DESCRIPTION
**Pull request by Reminisque**
This pull request addresses issue #1805.

I tested standard color depths from 8-bit to 32-bit. For 8-bit, the default mask was 2 bits for red, 3 bits for green, 2 bits for blue, and no bits for alpha. I only looked up color depths after starting this issue so please feel free to correct me if I am wrong, but is it supposed to be missing a bit for red? On wikipedia it had said the usual configuration is 3 bits for red and green and 2 bits for blue. As for 32-bit the default mask does not include bits for alpha, so I added another 32-bit depth surface with an included alpha mask.

Are there any test cases that I may be missing?

### CODE REVIEW

**Comment by Reminisque** [comment]
I am getting this error in CI:
```
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.15/lib/python2.7/site-packages/pygame-2.0.0.dev11-py2.7-linux-x86_64.egg/pygame/tests/surface_test.py", line 1272, in test_get_losses
    self.assertEqual(surf_8bit.get_losses(), (6, 5, 6, 8))
AssertionError: Tuples differ: (8, 8, 8, 8) != (6, 5, 6, 8)
First differing element 0:
8
6
- (8, 8, 8, 8)
?  ^  ^  ^
+ (6, 5, 6, 8)
?  ^  ^  ^
```
It was my mistake to assume that the losses will be the same for for a given color depth on all systems. However, I am unsure how to find out what losses I'll be given except when passing my own masks when creating a new Surface.

**Comment by MyreMylar** [comment]
> I tested standard color depths from 8-bit to 32-bit. For 8-bit, the default mask was 2 bits for red, 3 bits for green, 2 bits for blue, and no bits for alpha.

7 bits smacks of leaving space for 1 bit alpha to me (opaque or transparent) but since you said no alpha I may be wrong. Perhaps pygame uses 7 bits for colour to make it easy to transition from 8 bit without alpha to 8-bit with alpha with no colour alteration.

> It was my mistake to assume that the losses will be the same for for a given color depth on all systems. However, I am unsure how to find out what losses I'll be given except when passing my own masks when creating a new Surface. Should I pass a mask for all surfaces, except the one with no depth parameter as that one seems to work fine?

656 is another one of those compressed colour formats a bit like the old DXT that used to be popular. 

If the losses are different on CI compared to local I suspect you can figure out what the losses are likely to be from the colour format of the default display surface (returned from `set_mode()`)? I recall that mapped colours have to do with the display surface and that this is different between CI and most local configurations.

Perhaps also try setting the display driver to 'dummy' locally and see if you can reproduce the behaviour. I'll have a poke at that myself.

**Comment by Reminisque** [comment]
```
surf_set_mode = pygame.display.set_mode((100, 100))
surf_default = pygame.Surface((100, 100))
surf_8bit = pygame.Surface((100, 100), depth=8)
surf_8bit_332 = pygame.Surface((100, 100), depth=8, masks=rgba8)
surf_15bit = pygame.Surface((100, 100), depth=15)
surf_16bit = pygame.Surface((100, 100), depth=16)
surf_24bit = pygame.Surface((100, 100), depth=24)
surf_32bit_rgb = pygame.Surface((100, 100), depth=32)
surf_32bit_rgba = pygame.Surface((100, 100), depth=32, masks=rgba32)

print(surf.get_losses())
print(surf_default.get_losses())
print(surf_8bit.get_losses())
print(surf_8bit_332.get_losses())
print(surf_15bit.get_losses())
            print(surf_16bit.get_losses())
            print(surf_24bit.get_losses())
            print(surf_32bit_rgb.get_losses())
            print(surf_32bit_rgba.get_losses())
```

**Comment by Reminisque** [comment]
Is there a way to print out values in CI? I tried to see what losses I am getting in the CI, but am unable to get any output from the print statements I added to the test. The losses seem to be different from what I get locally with the video driver set to dummy, but I can only tell when a test fails so I would only get test output per commit.

Also it seems that alphas aren't valid for 8-bit and 15-bit. I tried to put 1 bit for alpha by removing 1 bit from blue to keep the depth the same, but it only worked locally. I haven't tried 32-bit yet on the CI, since it's one of the last tests and the failures come before it.

**Comment by Reminisque** [comment]
I got the losses to work. Only the 8-bit depth surface with no masks given is returning losses of 8 bits for all color components. Every other surface is returning losses that match what I get locally without setting the driver to dummy.

File: test/surface_test.py [hunk 1]
```diff
@@ -1240,19 +1239,55 @@ def todo_test_get_locks(self):
...
+            elif (display_surf.get_losses() == (8, 8, 8, 8)):
+                self.assertEqual(losses, (8, 8, 8, 8))
+
+            self.assertEqual(surf_8bit.get_losses(), (5, 5, 6, 8))
+            self.assertEqual(surf_15bit.get_losses(), (3, 3, 3, 8))
+            self.assertEqual(surf_16bit.get_losses(), (3, 2, 3, 8))
+            self.assertEqual(surf_24bit.get_losses(), (0, 0, 0, 8))
+            self.assertEqual(surf_32bit.get_losses(), (0, 0, 0, 0))
+            
+            # Method should fail when display is not intialized
```
Comment (MyreMylar) [inline test/surface_test.py:1284]: A small typo here in the word initialized/initialised.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Overall it looks good, test passes locally for me and it all makes sense I'll approve it with the small typo fixed.

**Comment by Reminisque** [comment]
I've added fixes for the typo and the extra parentheses. Looks like a bit of javascript syntax got mixed in while I was working on it.

Is it okay for the appveyor check to fail? I looked at what the error message, but there wasn't any indication that it was an error being raised during testing. I think it was a 403 HTTP error code.

**Review by MyreMylar — APPROVED** [review]

**Pull request merged**